### PR TITLE
examples: expand storage smartcontract with find() usage example

### LINKED ISSUE #364: Expand storage smartcontract with find() usage example
`examples/storage/storage.go`:
```
// TODO: storage.Find()
```
Making a separate issue as per #244.

### PR DESCRIPTION
**Pull request by AnnaShaleva**
closes #364

### CODE REVIEW

File: examples/storage/storage.go [hunk 1]
```diff
@@ -33,7 +34,24 @@ func Main(operation string, args []interface{}) interface{} {
...
 
-	// TODO: storage.Find()
+	// Returns an array of key-value pairs with key that matched the passed value
+	if operation == "find" {
+		if checkArgs(args, 1) {
+			value := args[0].([]byte)
+			iter := storage.Find(ctx, value)
+
+			result := []string{}
+			for {
```
Comment (fyrchik) [inline examples/storage/storage.go]: Does `for iterator.Next(iter) {` form work?

Comment (AnnaShaleva) [reply examples/storage/storage.go]: Yes, it does, fixed.

File: pkg/compiler/syscall.go [hunk 2]
```diff
@@ -87,5 +87,7 @@ var syscalls = map[string]map[string]string{
 		"Key":    "Neo.Iterator.Key",
 		"Keys":   "Neo.Iterator.Keys",
 		"Values": "Neo.Iterator.Values",
+		"Value":  "Neo.Iterator.Value",
+		"Next":   "Neo.Iterator.Next",
```
Comment (roman-khimov) [inline pkg/compiler/syscall.go]: Please keep 'em sorted.

File: pkg/interop/iterator/iterator.go [hunk 3]
```diff
@@ -26,3 +26,14 @@ func Keys(it Iterator) []interface{} {
 func Values(it Iterator) []interface{} {
 	return nil
 }
+
+// Next advances the iterator, return true if it is was successful
+// and false otherwise.
+func Next(it Iterator) bool {
```
Comment (roman-khimov) [inline pkg/interop/iterator/iterator.go]: Same thing here.

**Pull request merged**